src/app/dataset-view/additional-components/data-component/data-component.ts

```diff
@@ -27,14 +27,18 @@ import { sqlEditorOptions } from "src/app/dataset-block/metadata-block/component
 })
 export class DataComponent extends BaseComponent implements OnInit {
     @Input() public datasetBasics?: DatasetBasicsFragment;
-    @Output() public runSQLRequestEmit = new EventEmitter<string>();
+    @Output() public runSQLRequestEmit = new EventEmitter<DatasetRequestBySql>();
+
     public sqlEditorOptions = sqlEditorOptions;
     public savedQueries = DataTabValues.savedQueries;
     public sqlRequestCode = `select\n  *\nfrom `;
-
     public sqlErrorMarker: MaybeNull<string> = null;
     public currentSchema: MaybeNull<DatasetSchema> = null;
     public currentData: DataRow[] = [];
+    public isAllDataLoaded = false;
+
+    private skipRows: number;
+    private rowsLimit: number;
```

Comment:
I guess `rowsLimit` could have an initial value `AppValues.SQL_QUERY_LIMIT`.

While `skipRows` is intentionally `undefined`, so perhaps the type should be `MaybeUndefined<number>`. Right now the type annotation `number` is not entirely correct, since it can be `undefined` for a longer time.

Reply:
Changed as described